Source/WebCore/dom/Document.cpp

```diff
@@ -9020,6 +9020,16 @@ const Style::CustomPropertyRegistry& Document::customPropertyRegistry() const
     return styleScope().customPropertyRegistry();
 }
 
+const CSSCounterStyleRegistry& Document::counterStyleRegistry() const
+{
+    return styleScope().counterStyleRegistry();
```

Comment:
Is putting the registry in the `styleScope` the idiomatic way of doing this? For comparison, the document owns a `CSSFontSelector` directly. Maybe @anttijk knows?

Did you do this because it solves the lifetime problems of invalidating document state and rebuilding document state?

Reply:
According to Antti, there was no StyleScope back in the time so things were being added directly to the document that got too extensive. I've moved it here as he did the same for CustomProperty.

Reply:
Yea, it is fine. Style related things should generally go to style scope. Also many other things in Document should be factored out to reduce bloat.

Shadow scopes also potentially require their own registration though the CSSWG has not specced anything that requires it yet.